src/SubdomainNavBar/SubdomainNavBar.tsx

```diff
@@ -0,0 +1,373 @@
+import React, {useState, useRef, PropsWithChildren, forwardRef} from 'react'
+import clsx from 'clsx'
+import {ChevronLeftIcon, MarkGithubIcon, SearchIcon, XIcon} from '@primer/octicons-react'
+
+import {Button, FormControl, Heading, Text, TextInput} from '..'
+import {NavigationVisbilityObserver} from './NavigationVisbilityObserver'
+
+/**
+ * Design tokens
+ */
+import '../../lib/design-tokens/css/tokens/functional/components/subdomain-nav-bar/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './SubdomainNavBar.module.css'
+import {useOnClickOutside} from '../hooks/useOnClickOutside'
+
+export type SubdomainNavBarProps = {
+  /**
+   * Valid child elements are `SubdomainNavBar.Link`, `SubdomainNavBar.PrimaryAction`,
+   * `SubdomainNavBar.SecondaryAction` and `SubdomainNavBar.Search`
+   */
+  children?:
+    | React.ReactNode
+    | React.ReactElement<LinkProps>
+    | React.ReactElement<SearchProps>
+    | React.ReactElement<CTAActionProps>
+  /**
+   * Forward a custom HTML class attribute
+   */
+  className?: string
+  /**
+   * Fixes the navigation bar to the top of the viewport. Defaults to `true`.
+   */
+  fixed?: boolean
+  /**
+   * The title or name of the subdomain. Appears adjacent to the logo and is required for communicating content to assisitive technologies.
+   */
+  title: string
+  /**
+   * Optionally change the URL of the logo
+   */
+  logoLink?: string
```

Comment:
Nitpick: I wonder if this should be called `logoUrl` or `logoHref`? I initially thought it would accept a `Link` component from a routing library like React Router or Next

Reply:
I've changed to `logoHref` to be consistent with the `href` props on primary and secondary cta props